cpp/oneapi/dal/algo/shortest_paths/backend/cpu/traverse_default_kernel_nrh.cpp

```diff
@@ -80,7 +86,13 @@ inline void relax_edges_with_pred(const Topology& t,
         const EdgeValue new_dist = dp[u].dist + v_w;
         if (new_dist < old_dp.dist) {
             dp[v] = dist_pred(new_dist, u);
-            std::int64_t dest_bin = new_dist / delta;
+            ONEDAL_ASSERT(new_dist > 0);
+            ONEDAL_ASSERT(delta > 0);
+            ONEDAL_ASSERT(new_dist / delta <=
+                          static_cast<EdgeValue>(std::min(std::numeric_limits<std::int64_t>::max(),
+                                                          std::numeric_limits<EdgeValue>::max())));
```

Comment:
Why don't you use separate `constexpr` value to store max distance?

Reply:
It is used only in asserts that is in debug mode. Complicating code with ifdef assert does not cost such perf improvements for debugging.